src/store/modules/nft.js

```diff
@@ -413,6 +414,82 @@ const actions = {
     }
   },
   async fetchNFTClassAggregatedInfo({ commit, dispatch }, classId) {
+    const promise = this.$api.$get(api.getNFTClassMetadata(classId));
+    if (Array.isArray(classId)) {
+      const classesData = await promise;
+      const userPromises = [];
+      Object.entries(classesData).forEach(([classId, data]) => {
+        commit(TYPES.NFT_SET_NFT_CLASS_AGGREGATED_PROMISE, {
+          classId,
+          promise,
+        });
+        const { classData, iscnData, ownerInfo, listings, purchaseInfo } = data;
+        const iscnId = classData.parent.iscn_id_prefix;
+        commit(TYPES.NFT_SET_NFT_CLASS_METADATA, {
+          classId,
+          metadata: classData,
+        });
+        commit(TYPES.NFT_SET_ISCN_METADATA, { iscnId, data: iscnData });
+        commit(TYPES.NFT_SET_NFT_CLASS_OWNER_INFO, {
+          classId,
+          info: ownerInfo,
+        });
+        commit(TYPES.NFT_SET_NFT_CLASS_LISTING_INFO, {
+          classId,
+          info: listings,
+        });
+        // skip for non Writing NFT
+        if (purchaseInfo) {
+          commit(TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO, {
+            classId,
+            info: purchaseInfo,
+          });
+        }
+        if (classData.iscn_owner) {
+          const userPromise = dispatch(
+            'lazyGetUserInfoByAddress',
+            classData.iscn_owner
+          );
+          userPromises.push(userPromise);
+        }
+      });
+      // Need to await if the action fires in during SSR
+      if (!process.client) await userPromise;
+    } else {
+      commit(TYPES.NFT_SET_NFT_CLASS_AGGREGATED_PROMISE, { classId, promise });
+      const {
+        classData,
+        iscnData,
+        ownerInfo,
+        listings,
+        purchaseInfo,
+      } = await promise;
+      const iscnId = classData.parent.iscn_id_prefix;
+      commit(TYPES.NFT_SET_NFT_CLASS_METADATA, {
+        classId,
+        metadata: classData,
+      });
+      commit(TYPES.NFT_SET_ISCN_METADATA, { iscnId, data: iscnData });
+      commit(TYPES.NFT_SET_NFT_CLASS_OWNER_INFO, { classId, info: ownerInfo });
+      commit(TYPES.NFT_SET_NFT_CLASS_LISTING_INFO, { classId, info: listings });
+      // skip for non Writing NFT
+      if (purchaseInfo) {
+        commit(TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO, {
+          classId,
+          info: purchaseInfo,
+        });
+      }
+      if (classData.iscn_owner) {
+        const userPromise = dispatch(
+          'lazyGetUserInfoByAddress',
+          classData.iscn_owner
+        );
+        // Need to await if the action fires in during SSR
+        if (!process.client) await userPromise;
+      }
+    }
+  },
+  async fetchNFTClassesAggregatedInfo({ commit, dispatch }, classId) {
```

Comment:
Do we still keep `fetchNFTClassesAggregatedInfo` here?

Reply:
yes this is extra